src/components/Counter.component.js

```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+class Counter extends Component {
+  state = { count: this.props.initialState }
+  handleUp = () => {
+    this.setState({ count: this.state.count + 1 })
+  }
+  handleDown = () => {
+    this.setState({ count: this.state.count - 1 })
+  }
+  render() {
+    const { count } = this.state
+    return (
+      <div>
+        Count:
+        <data> {count} </data>
+        <button onClick={this.handleUp}> Increment </button>
+        <button onClick={this.handleDown}> Decrement </button>
+      </div>
+    )
+  }
+}
```

Comment:
so right now, if you update the initial count input, it won't update in this counter component. there are a couple ways of fixing that:

1. using [`componentWillReceiveProps`](https://reactjs.org/docs/react-component.html#componentwillreceiveprops)
2. moving the counter state and state updating functions up to the parent component

the former will register changes in incoming props which you can use to update your components state, but the latter will allow your counter component to not have any state at all (as it will live in the `App` component). that's preferred since you want to try to minimize state as much as possible. `Counter` will then become a stateless component, which is easier to reason about since its `render` is only a function of its `props`.

https://flaviocopes.com/react-beginners-guide/#moving-the-state-up-in-the-tree